capability.go

```diff
@@ -379,6 +379,10 @@ func (c *Client) IsSame(c2 *Client) bool {
 	return h1 == h2
 }
 
+// Done returns when the client begins shutting down.  The channel
+// returned by Done() is closed before ClientHook.Shutdown() is called.
```

Comment:
If we're going to expose this I think we need to explain the lifecycle semantics a bit better. Also skimming through it seems like this might interact with promise resolution in some ways that are not terribly obvious? Anyway it's a bad sign that I had to go dig through capability.go to work out what this actually does.